src/javascript/javascript-document.ts

```diff
@@ -68,7 +69,11 @@ export class JavaScriptDocument extends ParsedDocument<Node, Visitor> {
           continue;
         }
         if (callbackName in visitor) {
-          const result: VisitResult = visitor[callbackName](node, parent);
+          // TODO(rictic): is there a maintainable way to enforce the
```

Comment:
I don't think this is necessary or possible. We're off the type rails with this kind of meta programming.